app/routes.py

```diff
@@ -1,29 +1,31 @@
 from app import app
-from flask import render_template
+from flask import render_template, request
 import scripts.pains.LillyMedchemRules.pains as pains
-from app.data.smiles import smiles
-from app.data.smiles import smiles_with_murcko
+from app.data.smiles import smiles, construct_smiles, filter_smiles
 import scripts.clustering.clustering as clustering
 from app.data.clustering import get_smiles_json
 
 @app.route('/')
 @app.route('/index')
-def index():
+def index():     
   return render_template('index.html', title='Cheminformatic Analysis')
 
-@app.route('/')
-@app.route('/cluster')
-def cluster():
 
-  inputs = smiles.keys()
-  
-  bad_smiles = pains.get_bad_smiles_and_reasons(inputs)
+@app.route("/cluster", methods=['GET', 'POST'])
+def upload():
+    if request.method == 'POST':
+        try:
+            data = request.get_array(field_name='file')
+            construct_smiles(data)
+        except: 
+            return render_template('index.html', title='Cheminformatic Analysis', errors=["Please input a valid file format"])
 
-  tanimoto_smiles = clustering.add_tanimoto_coefficients(smiles)
-  
-  tanimoto_smile_strings = [" ".join([str(i) for i in v['similarities'].values()]) for v in tanimoto_smiles.values()]
+        inputs = smiles.keys()
+        
+        good_smiles = filter_smiles(pains.get_smiles(inputs))
 
-  get_smiles_json(tanimoto_smiles)
+        tanimoto_smiles = clustering.add_tanimoto_coefficients(good_smiles)
 
+        get_smiles_json(tanimoto_smiles)
 
-  return render_template('cluster.html', title='Cheminformatic Analysis', bad_smiles=bad_smiles, tanimoto_smiles=tanimoto_smile_strings)
\ No newline at end of file
+        return render_template('cluster.html', title='Cheminformatic Analysis')
```

Comment:
Are we planning on changing where this code goes later? Because it seems kind of odd to go straight from input to cluster.

Reply:
Yeah, probably! If we can add the manual PAINS selection then that would go before clustering.

Reply:
If you merge it now, then I can fit those next step changes into my branch.

Reply:
awesome! I'll merge